Description: Try to parse DEBIAN/control to get the packages version
 This patch changes the way isnative determines the version of a package.
 It first tries to parse debian/$package/DEBIAN/control and falls back to
 parsing the changelog.
 .
 Isnative has the side effect of setting $dh{VERSION}, which is used in
 udeb_filename. If dpkg-gencontrol -v<other version> is used then the
 version in DEBIAN/control and debian/changelog will disagree and without
 this patch udeb_filename will return the wrong filename and the build
 will fail in dpkg_genchanges.
 .
 The patch applies to debhelper from squeeze (8.0.0), squeeze-backports
 (9.20120419~bpo60+1), lucid (7.4.15) and precise (9.20120115ubuntu3).
Author: Goswin von Brederlow <brederlo@q-leap.de>
Last-Update: 2012-06-13

---
diff -Nru debhelper-8.0.0/Debian/Debhelper/Dh_Lib.pm debhelper-8.0.0+ql1/Debian/Debhelper/Dh_Lib.pm
--- debhelper-8.0.0/Debian/Debhelper/Dh_Lib.pm	2010-06-24 02:35:34.000000000 +0200
+++ debhelper-8.0.0+ql1/Debian/Debhelper/Dh_Lib.pm	2012-06-13 12:35:51.000000000 +0200
@@ -437,11 +437,16 @@
 			$isnative_changelog="debian/changelog";
 		}
 		# Get the package version.
-		my $version=`dpkg-parsechangelog -l$isnative_changelog`;
+		my $version=`grep 2>/dev/null "^Version:" "debian/$package/DEBIAN/control"`;
 		($dh{VERSION})=$version=~m/Version:\s*(.*)/m;
-		# Did the changelog parse fail?
+		# Did the DEBIAN/control parse fail?
 		if (! defined $dh{VERSION}) {
-			error("changelog parse failure");
+			$version=`dpkg-parsechangelog -l$isnative_changelog`;
+			($dh{VERSION})=$version=~m/Version:\s*(.*)/m;
+			# Did the changelog parse fail?
+			if (! defined $dh{VERSION}) {
+				error("changelog parse failure");
+			}
 		}
 
 		# Is this a native Debian package?
